### ISSUE TITLE
Post-block compute-dtype round (reconstruct then FP16/BF16)

### ISSUE THREAD

**Issue by SamirMoustafa** [SamirMoustafa/torch-floating-point#18]
## Problem

`block_round` reconstructs \(y = (e-z)\,s\,s_g\) in the working dtype (typically FP32). That overstates devices that **decompress then GEMM in FP16/BF16**. E2M3 max \(7.5\) × UE8M0 max \(2^{127}\) is far outside FP16 (\(65504\)); a QAT run that never saturates to FP16 is not Qualcomm’s path.

Eval-only `Round(fp16)(BlockRound(spec)(x).detach())` works. With gradients, a second `Round` on the same tensor raises `Double quantization detected.` Nested `Round` is banned on purpose; the compute snap must live **inside** `block_round`.

## Exists today

`block_round` returns FP32 (or the input dtype). `s_global` is a tensor-level scale, not a compute codebook. `rounder=` replaces the **element** map, not a post-reconstruct IEEE-16 snap.

## Proposal

```python
BlockRound(spec, compute_fp=fp16)(x)
```

- `compute_fp is None` (default): current behavior.
- Else: after reconstruct, snap \(y\) with `Round(compute_fp)` **without** a second STE on \(x\). Clipped STE stays on the **element** range; the compute snap is forward-only / STE identity (hardware saturates, then GEMM).
- `compute_fp` is a `FloatingPoint` (IEEE FP16 / BF16), not `torch.float16`.

Start with **product-then-round**. Gaudi’s naive kernel casts **both** \(e\) and \(s\) to BF16 then multiplies (1 ULP vs product-then-round); document that variant only if a fixture needs it.

Do **not** export `MXFP6`. Fixture: OCP MXFP6 E2M3 + UE8M0 + `compute_fp=fp16` (caller `block_size`; do not invent QAIC \(k\)).

## Unblocks

- Qualcomm Cloud AI 100: MXFP6 is DRAM compression (`-mxfp6-matmul`); vector-engine decompress, NSP compute stays FP16 — not fused MX MMA. https://quic.github.io/cloud-ai-sdk-pages/latest/blogs/Microscaling/microscaling/
- Intel Gaudi: `dequant_block_fp8_weight_naive` does `weight.to(dtype) * weight_scale.to(dtype)` then `F.linear`. Native `fp8_gemm_v2` is fused — skip here. https://github.com/vllm-project/vllm-gaudi/blob/main/vllm_gaudi/extension/ops.py
- NVIDIA TE: `nvte_dequantize` materializes a compute dtype. Blackwell MMA applies \(e\cdot s\) in the tensor core — do not model that. https://docs.nvidia.com/deeplearning/transformer-engine/user-guide/api/c/cast.html
- Graphcore: PopRT `--precision fp8_weight` is FP8 store, FP16 infer. `fp8Scale` is per-tensor, not MX block \(k\). https://docs.graphcore.ai/projects/poprt-user-guide/en/latest/features/fp8.html

## Acceptance

- Block amax with \(e\cdot s > 65504\): `compute_fp=fp16` saturates to FP16 max; without, it does not.
- `BlockRound(spec, compute_fp=fp16)(x).sum().backward()` does **not** raise `Double quantization detected.`
- Nested `Round(fp16)(BlockRound(spec)(x))` still raises (ban stays).
- Docs recipes for the four Unblocks paths. No fused-MMA claim.

## Out of scope

Fused MX/FP8 GEMM (cuBLAS, Blackwell Tensor Core, Gaudi `fp8_gemm_v2`, IPU native FP8 MatMul). QAIC unpublished \(k\). No `MXFP6` export.

**Comment by SamirMoustafa** [comment]
Explorer: if this ships, `docs/javascripts/fp-codec.js` reconstruct (and goldens) need the same post-block `compute_fp` snap, plus an explorer knob — otherwise the page stays FP32 while Python saturates to FP16/BF16.
